src/task/MonitorTask.ts

```diff
@@ -79,12 +85,28 @@ export default class MonitorTask
     this.context.realtimeController.realtimeSubscribeToLocalSignalStrengthChange(
       this.checkAndSendWeakSignalEvent
     );
+    this.startTimestampMs = Date.now();
+    this.attendeeId = this.context.meetingSessionConfiguration.credentials.attendeeId;
+    this.context.realtimeController.realtimeSubscribeToAttendeeIdPresence(
+      this.realtimeAttendeeIdPresenceHandler
+    );
 
     this.context.connectionMonitor.start();
     this.context.statsCollector.start(this.context.signalingClient, this.context.videoStreamIndex);
     this.context.signalingClient.registerObserver(this);
   }
 
+  publishEvent(): void {
```

Comment:
I don't think you need to create an extra function, since it's used only once. You just need the following code inline using optional chaining (?).

```
this.context.eventController?.publishEvent('attendeePresenceReceived', {
  attendeePresenceDurationMs: this.context.attendeePresenceDurationMs
});
```

Reply:
Done, used it inline